packages/actor-http-fetch/lib/ActorHttpFetch.ts

```diff
@@ -58,22 +58,44 @@ export class ActorHttpFetch extends ActorHttp {
       action.init.headers = ActorHttp.headersToHash(action.init.headers);
     }
 
+    const requestInit = { ...action.init };
+
+    if (action.context.get(KeysHttp.includeCredentials)) {
+      requestInit.credentials = 'include';
+    }
+
+    const httpTimeout: number | undefined = action.context?.get(KeysHttp.httpTimeout);
+    let setupHttpTimeout: (() => void) = () => {
+      // No op by default
+    };
+    if (httpTimeout !== undefined) {
+      // Fallback to abort-controller for Node 14 backward compatibility
+      /* istanbul ignore next */
+      const AbortController = global.AbortController || await import('abort-controller');
+      const controller = new AbortController();
+      requestInit.signal = controller.signal;
+      setupHttpTimeout = () => {
+        setTimeout(() => controller.abort(), httpTimeout);
```

Comment:
We need to save the timeout handler, and clear it if the response comes in before the timeout.
Otherwise, the process will hang if queries end faster than this timeout.

Reply:
Genuine question: Are you sure? The "should work with a timeout" test seems to pass even if the response comes in before the timeout. I also made some real world hand tests that did not exposed this problem either. But I'm maybe missing something.

Reply:
I'm pretty sure, yes. I remember I had to do similar things in the past when working with timeouts.
Unless something changed in Node recently, but I can't find any news on that AFAICS.

You could try by setting the timeout to a very high value (e.g. 1 minute). If you then issue a query, then the query results will arrive quickly, but the process will only end after one minute, while it is allowed to exit earlier.

Reply:
Indeed, my bad.  Thank you!